orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pollers/RestorePinnedServerGroupsPoller.java

```diff
@@ -0,0 +1,322 @@
+    this.triggeredCounter = registry.counter("poller.restorePinnedServerGroups.triggered");
+  }
+
+  @Override
+  protected long getPollingInterval() {
+    return TimeUnit.MINUTES.toSeconds(10);
+  }
+
+  @Override
+  protected String getNotificationType() {
+    return "restorePinnedServerGroups";
+  }
+
+  @Override
+  protected void startPolling() {
+    subscription = Observable
+      .timer(getPollingInterval(), TimeUnit.SECONDS, scheduler)
+      .repeat()
+      .filter(interval -> tryAcquireLock())
+      .subscribe(interval -> {
+        try {
+          poll();
+        } catch (Exception e) {
+          log.error("Error checking for pinned server groups", e);
+        }
+      });
+  }
+
+  void poll() {
+    log.info("Checking for pinned server groups");
+
+    List<PinnedServerGroupTag> pinnedServerGroupTags = fetchPinnedServerGroupTags();
+    log.info("Found {} pinned server groups", pinnedServerGroupTags.size());
+
+    if (pinnedServerGroupTags.isEmpty()) {
+      return;
+    }
+
+    pinnedServerGroupTags = pinnedServerGroupTags.stream()
+      .filter(this::hasCompletedExecution)
+      .collect(Collectors.toList());
+
+    log.info("Found {} pinned server groups with completed executions", pinnedServerGroupTags.size());
+
+    for (PinnedServerGroupTag pinnedServerGroupTag : pinnedServerGroupTags) {
+      try {
+        ServerGroup serverGroup = fetchServerGroup(
+          pinnedServerGroupTag.account,
+          pinnedServerGroupTag.location,
+          pinnedServerGroupTag.serverGroup
+        );
+
+        List<Map<String, Object>> jobs = new ArrayList<>();
+        jobs.add(buildDeleteEntityTagsOperation(pinnedServerGroupTag, serverGroup));
+
+        if (serverGroup.capacity.min.equals(pinnedServerGroupTag.pinnedCapacity.min)) {
+          jobs.add(0, buildResizeOperation(pinnedServerGroupTag, serverGroup));
+
+          // ensure that the tag cleanup comes after the resize operation has completed
+          jobs.get(1).put("requisiteStageRefIds", Collections.singletonList(jobs.get(0).get("refId")));
+        }
+
+        Map<String, Object> cleanupOperation = buildCleanupOperation(pinnedServerGroupTag, serverGroup, jobs);
+        log.info((String) cleanupOperation.get("name"));
+
+        // TODO-AJ Need to provide an authentication context such that resize operations in privileged accounts work!
+        executionLauncher.start(
+          Execution.ExecutionType.ORCHESTRATION,
+          objectMapper.writeValueAsString(cleanupOperation)
+        );
+
+        triggeredCounter.increment();
+      } catch (Exception e) {
+        log.error("Failed to unpin server group (serverGroup: {})", pinnedServerGroupTag.serverGroup, e);
+        errorsCounter.increment();
+      }
+    }
+  }
+
+  List<PinnedServerGroupTag> fetchPinnedServerGroupTags() {
+    List<EntityTags> allEntityTags = retrySupport.retry(() -> objectMapper.convertValue(
```

Comment:
Seen this before, I don't have any strong feelings either way, but should `retrySupport` really be wrapping calls to the object mapper?

Reply:
Not sure it particularly matters in this case ... more importantly it's wrapping the nested call to `oortService.getEntityTags()`.